src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -0,0 +1,357 @@
+          const state: any = getState() // ReduxState causes circular dependency
+          const ethPrecision = state.assets.byId[ethAssetId].precision
+          const lpAssetPrecision = state.assets.byId[foxEthLpAssetId].precision
+          const ethPrice = state.marketData.crypto.byId[ethAssetId].price
+          const ethersProvider = getEthersProvider()
+          const uniV2LPContract = new Contract(
+            UNISWAP_V2_WETH_FOX_POOL_ADDRESS,
+            IUniswapV2Pair.abi,
+            ethersProvider,
+          )
+          const pair = await fetchPairData(
+            new Token(0, WETH_TOKEN_CONTRACT_ADDRESS, 18),
+            new Token(0, FOX_TOKEN_CONTRACT_ADDRESS, 18),
+            Fetcher.fetchPairData,
+            ethersProvider,
+          )
+
+          const blockNumber = await ethersProvider.getBlockNumber()
+
+          const calculatedApy = await calculateAPRFromToken0({
+            token0Decimals: pair.token0.decimals,
+            token0Reserves: pair.reserve0,
+            blockNumber,
+            uniswapLPContract: uniV2LPContract,
+          })
+          const apy = bnOrZero(calculatedApy).div(100).toString()
+          const reserves = await uniV2LPContract.getReserves()
+          // Amount of Eth in liquidity pool
+          const ethInReserve = bnOrZero(reserves?.[0]?.toString()).div(`1e${ethPrecision}`)
+
+          // Total market cap of liquidity pool in usdc.
+          // Multiplied by 2 to show equal amount of eth and fox.
+          const totalLiquidity = ethInReserve.times(ethPrice).times(2)
+          const tvl = totalLiquidity.toString()
+          const totalSupply = await uniV2LPContract.totalSupply()
+          const price = bnOrZero(tvl)
+            .div(bnOrZero(totalSupply.toString()).div(`1e${lpAssetPrecision}`))
+            .toString()
+          const lpMarketData = {
+            [foxEthLpAssetId]: { price, marketCap: '0', volume: '0', changePercent24Hr: 0 },
+          }
+          // hacks for adding lp price and price history
+          dispatch(marketData.actions.setCryptoMarketData(lpMarketData))
+          Object.values(HistoryTimeframe).forEach(timeframe => {
+            dispatch(
+              marketData.actions.setCryptoPriceHistory({
+                data: [{ price: bnOrZero(price).toNumber(), date: 0 }],
+                args: { timeframe, assetId: foxEthLpAssetId },
+              }),
+            )
+          })
+          const data = { tvl, apy, isLoaded: true }
+          dispatch(foxEth.actions.upsertLpOpportunity(data))
+          return { data }
+        } catch (err) {
+          moduleLogger.error(err, 'getFoxEthLpGeneralData')
+          return {
+            error: {
+              error: 'LP general data is not loaded',
+              status: 'CUSTOM_ERROR',
+            },
+          }
+        }
+      },
+    }),
+    getFoxEthLpWalletData: build.query<GetFoxEthLpWalletDataReturn, GetFoxEthLpWalletDataArgs>({
```

Comment:
Same thing here, I think the terminology could be improved:

1. The `wallet` terminology carries the mental model of assets in wallet, which isn't the case when there is a delegated balance in a smart contract
2. With the multi-account feature currently in progress, we will not only handle account zero in a wallet, but more `accountNumber`s - using the `wallet` terminologies has implicit ties to the account zero architecture 

What about using the `accountData` terminology instead?